### PR TITLE
Add queryID in userAgent for spark delta sharing queries

### PR DESCRIPTION
**Pull request by linzhou-db**
Add queryID in userAgent for spark delta sharing queries

### CODE REVIEW

File: client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala [hunk 1]
```diff
@@ -1124,7 +1125,10 @@ case class DeltaSharingSource(
...
       Some(v)
     } else if (options.startingTimestamp.isDefined) {
-      Some(deltaLog.client.getTableVersion(deltaLog.table, options.startingTimestamp))
+      val version = deltaLog.client.getTableVersion(deltaLog.table, options.startingTimestamp)
+      logInfo(s"Got table version $version for timestamp ${options.startingTimestamp} " +
+        s"from Delta Sharing Server.")
+      Some(version)
     } else {
       None
     }
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:1134]: Can this change be unit tested?

Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:1134]: there's no real code change, just added a logInfo.

Comment (chakankardb) [reply client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:1134]: Aren't we changing the user agent header being sent by this client?
Can we test that the user agent as various fields? (perhaps a test was missing earlier).

Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:1134]: I see, taking a look.

Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:1135]: added

**Review by chakankardb — APPROVED** [review]

**Pull request merged**